### PR TITLE
[ticket/14440] Correctly remove the web root path in update_web_root_path

### PR DESCRIPTION
**Pull request by Nicofuma**
https://tracker.phpbb.com/browse/PHPBB3-14440

PHPBB3-14440

### CODE REVIEW

File: phpBB/phpbb/path_helper.php [hunk 1]
```diff
@@ -100,11 +100,17 @@ public function get_php_ext()
 	*/
 	public function update_web_root_path($path)
 	{
+		$web_root_path = $this->get_web_root_path();
+
+		if (strpos($path, $web_root_path) === 0)
```
Comment (VSEphpbb) [inline phpBB/phpbb/path_helper.php]: Maybe, since this is a bit of a cheat, add a comment explaining why this is done?

**Comment by marc1706** [comment]
@VSEphpbb can you confirm that this in fact fixes the issues described in that issue report?

**Comment by VSEphpbb** [comment]
@marc1706 i already did... the hope is that it doesnt break anything unforseen, but passing tests are a good sign.

**Comment by bantu** [comment]
Tests?

**Comment by Nicofuma** [comment]
Tests added

**Pull request merged**